WordPress/src/main/java/org/wordpress/android/ui/DeeplinkNavigator.kt

```diff
@@ -0,0 +1,32 @@
+package org.wordpress.android.ui
+
+import android.net.Uri
+import androidx.appcompat.app.AppCompatActivity
+import org.wordpress.android.ui.DeeplinkNavigator.NavigateAction.OpenInWebView
+import org.wordpress.android.ui.DeeplinkNavigator.NavigateAction.StartCreateSiteFlow
+import org.wordpress.android.util.UriWrapper
+import javax.inject.Inject
+
+class DeeplinkNavigator
+@Inject constructor() {
+    fun handleNavigationAction(navigateAction: NavigateAction, activity: AppCompatActivity) {
+        when (navigateAction) {
+            is OpenInWebView -> {
+                DeeplinkWebViewActivity.openUrl(activity, navigateAction.uri.toString())
+            }
+            is StartCreateSiteFlow -> {
+                if (navigateAction.isSignedToWpCom) {
+                    ActivityLauncher.newBlogForResult(activity)
+                } else {
+                    ActivityLauncher.addSelfHostedSiteForResult(activity)
+                }
```

Comment:
Two thoughts here:

- If the user is not logged in to WP.com, I think it would probably be better for them to be redirected to the WP.com login flow instead of the add site flow.
- In addition to that, we should also check if the user is not logged in at all, in which case I think they should be redirected to the full login flow.

Let me know what you think!

Reply:
hmm, good point but I think we should only redirect to the WP.com flow because the link in the email only comes to WP.com users and I don't see a usecase where we'd like to redirect a user who got the "Create a site" email to the self-hosted flow. WDYT?